pages/apply/overview.tsx

```diff
@@ -10,22 +10,38 @@ import SummaryList, {
 } from '../../components/summary-list';
 import Tag from '../../components/tag';
 import ApplicantName from '../../components/application/ApplicantName';
-import { Applicant } from '../../domain/HousingApi';
+import { Applicant, Application } from '../../domain/HousingApi';
 import { useAppDispatch, useAppSelector } from '../../lib/store/hooks';
 import { applicationStepsRemaining } from '../../lib/utils/resident';
 import {
   sendConfirmation,
   completeApplication,
   sendDisqualifyEmail,
+  loadApplication,
+  importApplication,
 } from '../../lib/store/application';
 import { checkEligible } from '../../lib/utils/form';
 import withApplication from '../../lib/hoc/withApplication';
 import Custom404 from '../404';
+import { GetServerSideProps } from 'next';
+import { getApplication } from '../../lib/gateways/applications-api';
+import { getUser } from '../../lib/utils/users';
+import { useEffect } from 'react';
 
-const ApplicationPersonsOverview = (): JSX.Element => {
+interface PageProps {
+  data: Application;
+}
+
+const ApplicationPersonsOverview = ({data}: PageProps): JSX.Element => {
   const router = useRouter();
   const dispatch = useAppDispatch();
 
+  useEffect(() => {
+    dispatch(
+      importApplication(data)
+    );
+  }, []);
```

Comment:
So I see the approach that you're going for here - preload the application with the page load and then put it straight into the store at the first render.

Redux does actually let you pre-fill the store when you create it, it's just that we're initing the store in the App rather than on each page. This data that you're loading has wider context than just this page though - so perhaps you need a getServerSideProps for the app and then you can take care of the whole thing there.

Reply:
Ok thanks, have switched the code to load the application in the app init similar to how it worked before.